components/discord_bot/actions/common.mjs

```diff
@@ -0,0 +1,70 @@
+import maxBy from "lodash.maxby";
+import common from "../common.mjs";
+import constants from "../constants.mjs";
+
+const { discord } = common.props;
+
+export default {
+  ...common,
+  methods: {
+    async paginateMessages({
+      $, channelId, before, after, around,
+      limit = constants.DEFAULT_PAGE_LIMIT,
+      max = constants.DEFAULT_MAX_ITEMS,
+    }) {
+      const hasFilter = before || after || around;
+      let messages = [];
+      let nextMessages = [];
+      let lastId;
+
+      do {
+        const lastLimit = max - messages.length;
+
+        const params = !messages.length && hasFilter
+          ? {
+            limit,
+            before,
+            after,
+            around,
+          }
+          : {
+            limit: lastLimit < limit
+              ? lastLimit
+              : limit,
+            before: lastId,
+          };
```

Comment:
I think `after` should be included here as well so that we don't accidentally go past the message/item specified by `after` in case it's not on the first page of results.

Reply:
I didn't quite get what you meant here. Since `before`, `after` and `around` are mutually exclusive I can't use `before` and `after` at the same time. And for me It would make more sense if I paginate using `before` because I will always get the latest message first so I will be paginating backwards in time.

Reply:
Oh yeah, you’re right. We do want to check that we’re not going past the after ID though. Can we check for it at the bottom of the loop, or as the messages are added to the array? Once we’ve reached the after ID, we don’t need to continue the pagination.